File: utils/ukamacli/cmd/restart_node.go
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * Copyright (c) 2023-present, Ukama Inc.
 */

package cmd

import (
	"github.com/spf13/cobra"

	"github.com/ukama/ukama/utils/ukamacli/internal/resource/node"
)

var RestartNodeCmd = &cobra.Command{
	Use:   "node",
	Short: "Restart node",
	Long:  `The Restart node command Restarts the given Ukama node.`,

	Aliases:      []string{"n"},
	SilenceUsage: true,

	RunE: func(cmd *cobra.Command, args []string) error {
		return node.Restart()
	},
}

func init() {
	restartCmd.AddCommand(RestartNodeCmd)
}

File: systems/console-bff/controller/resolvers/restartNode.ts
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * Copyright (c) 2023-present, Ukama Inc.
 */
import { Arg, Ctx, Mutation, Resolver } from "type-graphql";

import { CBooleanResponse } from "../../common/types";
import { Context } from "../context";
import { RestartNodeInputDto } from "./types";

@Resolver()
export class RestartNodeResolver {
  @Mutation(() => CBooleanResponse)
  async restartNode(
    @Arg("data") data: RestartNodeInputDto,
    @Ctx() ctx: Context
  ): Promise<CBooleanResponse> {
    const { dataSources, baseURL } = ctx;
    return dataSources.dataSource.restartNode(baseURL, data);
  }
}
